### ISSUE TITLE
iOS background service is not working. Error: Could not schedule app refresh: Error Domain=BGTaskSchedulerErrorDomain Code=3 "(null)"

### ISSUE THREAD

**Issue by lahirunc** [ekasetiawans/flutter_background_service#254]
I have followed everything on ReadMe and running the app on a physical device (iOS 16.0.2) but Im getting following error.

`Could not schedule app refresh: Error Domain=BGTaskSchedulerErrorDomain Code=3 "(null)"`

**AppDelegate.swift**
![image](https://user-images.githubusercontent.com/20990764/204696745-03cb20ac-9a61-480b-a601-1383aac09c05.png)

**Info.plist**
![image](https://user-images.githubusercontent.com/20990764/204696864-9ab219b0-aa03-4af9-b547-c6760f125f4a.png)
![image](https://user-images.githubusercontent.com/20990764/204696891-af0bd3d6-3a67-4d51-b3f3-0e9c6a924b50.png)

**App permissions - iOS**
<img width="300" alt="image" src="https://user-images.githubusercontent.com/20990764/204697487-2b2c97de-df6b-47ec-ba2f-60ea45f741c1.png">

The background service works well in android. Can someone tell me what am I doing wrong?

Thanks. 😊

**Comment by ekasetiawans** [comment]
You have to change “your.custom.task.identifier” that must be match with BGTaskSchedullerPermittedIdentifier in your Info.plist

**Comment by lahirunc** [comment]
@ekasetiawans I did that too and i'm getting this error and app terminates itself

**AppDelegate.swift**
![Screenshot 2022-11-30 at 2 31 22 pm](https://user-images.githubusercontent.com/20990764/204701108-cb8c1290-106f-4974-8316-8aa10e1e9cfd.png)

**Error**
`* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGKILL
    frame #0: 0x0000000211c5fb48 libsystem_kernel.dylib`mach_msg2_trap + 8
libsystem_kernel.dylib`mach_msg2_trap:
->  0x211c5fb48 <+8>: ret
libsystem_kernel.dylib`macx_swapon:
    0x211c5fb4c <+0>: mov    x16, #-0x30
    0x211c5fb50 <+4>: svc    #0x80
    0x211c5fb54 <+8>: ret
Target 0: (Runner) stopped.
Lost connection to device.`

**Comment by ekasetiawans** [comment]
is it happened for the example project too?

**Comment by lahirunc** [comment]
@ekasetiawans I didnt try the example project.

**Comment by lahirunc** [comment]
@ekasetiawans Any update on this?

**Comment by anaya12345** [comment]
I am also having the similar issue. My background service is not working on iOS. 
Please give the solution.

**Comment by ced1check** [comment]
Same issue here, background service is never called.
Pressing 'Simulate Background Fetch' results in message 'Background handler is disabled' and nothing else happens.
Not sure what to do to fix this, or use a different plugin.

**Comment by EXEIdeas** [comment]
Yes, Same issue here, Still not working in IOS and `onIosBackground()` is never called...

Rest can I start server from service as I am getting error and thinking to restart service after each 30 sec n `onStart()`...???

`[BackgroundTask] Background Task 1 ("Flutter debug task"), was created over 30 seconds ago. In applications running in the background, this creates a risk of termination. Remember to call UIApplication.endBackgroundTask(_:) for your task in a timely manner to avoid this.
`

**Comment by jrmanes** [comment]
any updates about this topic?, I'm having the same issue

**Comment by EXEIdeas** [comment]
> any updates about this topic?, I'm having the same issue
> 
> if you finally found the solution, would you main to share it?

@jrmanes Still Waiting... :-(

**Comment by jrmanes** [comment]
> > any updates about this topic?, I'm having the same issue
> > if you finally found the solution, would you main to share it?
> 
> @jrmanes What is your Main issue? iOS Background is not working completely or stop working after 30 sec?

it's weird, I could make it work in the simulator (deploying it with vscode and from the xcode, both work), but when I tried to deploy it via testflight in a device (iPhone), it does nothing...
does it make sense to you?

**Comment by anaya12345** [comment]
Yes, it not working for iOS, need to write Swift code for location tracking

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by EXEIdeas** [comment]
> > > any updates about this topic?, I'm having the same issue
> > > if you finally found the solution, would you main to share it?
> > 
> > 
> > @jrmanes What is your Main issue? iOS Background is not working completely or stop working after 30 sec?
> 
> it's weird, I could make it work in the simulator (deploying it with vscode and from the xcode, both work), but when I tried to deploy it via testflight in a device (iPhone), it does nothing... does it make sense to you?

Are you using `@pragma('vm:entry-point')` ?

**Comment by EXEIdeas** [comment]
Thanks for the update. Good to hear that it is working for you. Well, you are talking about an iOS background service that is terminated after 30 sec by default by iOS itself so this plugin can't control it. Well, only this premium plugin can do the work even in the background continuously after the app closes. https://www.transistorsoft.com/shop/products/flutter-background-geolocation.

**Comment by anaya12345** [comment]
Yes, it not working for iOS, need to write Swift code for location tracking


________________________________

**Comment by jrmanes** [comment]
hey @EXEIdeas @anaya12345 

so, that means that this plugin is not working at all?

my use case is, I need to schedule a task every 1-5h (I don't need to do it in real time), to do so, theo only way is, either I go with the plugin that you mentioned from transistorsoft.com or with swift code?, is not a way to do so with this plugin?

**Comment by jrmanes** [comment]
if `swift` is the only way, do you have any recommendation/example to follow @anaya12345 ?

**Comment by EXEIdeas** [comment]
> hey @EXEIdeas @anaya12345
> 
> so, that means that this plugin is not working at all?
> 
> my use case is, I need to schedule a task every 1-5h (I don't need to do it in real time), to do so, theo only way is, either I go with the plugin that you mentioned from transistorsoft.com or with swift code?, is not a way to do so with this plugin?

You can use this as by default this will work each after about 15 min for 30 sec only...

**Comment by jrmanes** [comment]
> > hey @EXEIdeas @anaya12345
> > so, that means that this plugin is not working at all?
> > my use case is, I need to schedule a task every 1-5h (I don't need to do it in real time), to do so, theo only way is, either I go with the plugin that you mentioned from transistorsoft.com or with swift code?, is not a way to do so with this plugin?
> 
> You can use this as by default this will work each after about 15 min for 30 sec only...

okay, so technically it should work for iOS in release mode too, max every 15' and for 30", but it should work, have you been able to make it work for iOS?, just to confirm it

**Comment by anaya12345** [comment]
This plugin work for Android well.
Also works for iOS of user have launched app and device is in unlock state and not in sleep mode.

To work bg service when device is sleep also, I used the swift code to fetch continuous location updates, there is requirement of location tracking in my application.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by EXEIdeas** [comment]
Yes, To get Continuous Location in BG of iOS, you need to use Swift code and in this process, you can make your application live and active and can do manythings extra too...

**Comment by anaya12345** [comment]
[like]  Anaya Joshi reacted to your message:
________________________________

**Comment by DhananjayChawdaMMF** [comment]
BGTaskSchedulerErrorDomain Code=3

The error occures because you give the differnece between the id you given in the AppDelegate file and info.plist 

you need to give same id on both place 

SwiftFlutterBackgroundServicePlugin.taskIdentifier = "dev.flutter.background.refresh"

<key>BGTaskSchedulerPermittedIdentifiers</key>
	<array>
		<string>dev.flutter.background.refresh</string>
	</array>
